pkg/virt-launcher/virtwrap/cmd-server/server.go

```diff
@@ -63,7 +63,7 @@ func (s *Launcher) Sync(args *cmdclient.Args, reply *cmdclient.Reply) error {
 		return nil
 	}
 
-	_, err = s.domainManager.SyncVMI(vmi, s.allowEmulation)
+	_, err = s.domainManager.SyncVMI(vmi, s.useEmulation)
```

Comment:
can't we determine if useEmulation should be used or not based on the vmi requests field?  It doesn't seem like the cmd server needs to know about useEmulation before hand

Reply:
> I think we were specifically trying to avoid any reference to hardware acceleration inside the VM/VMI specs, hence the cluster-wide setting.

is there a way to detect within the virt-launcher pod whether the kvm device is present or not?

Reply:
No. The controller - cluster level - decides to either add or not add the kvm device. The operator can configure the cluster to use emulation if no hardware acceleration is available. This was the agreement we came to.

Reply:
@davidvossel a little context: this is a side effect of the scheduling contract with device plugins. Basically a node advertises "yes. I have KVM devices" (these show up as namespaced entities in the `allocatable` and `capacity` sections of the node's yaml. e.g. `devices.kubevirt.io/kvm: "110"`). Thus when virt-controller makes the decision to add a request for a KVM device to a pod, that pod can now *only* be scheduled on nodes which provide KVM devices.

Or stated another way, the decision to request a KVM device basically mandates that one will be present (or the other outcome is that the pod will hang out in scheduling limbo forever). This motivated the change of semantics from "allowEmulation" to "useEmulation" as the virt-controller can no longer take a passive role in the process.

Reply:
works for me :+1: